File: 190-reverse-bits/190-reverse-bits.java
public class Solution {
    // you need treat n as an unsigned value
    public int reverseBits(int n) {
        int m = 0;
        for (int i = 0; i < 32; i++) {
            m = m << 1;
            m += n % 2;
            n = n >>> 1;    
        }
        return m;
    }
}

File: 190-reverse-bits/190-reverse-bits.kt
class Solution {
    fun reverseBits(num:Int):Int {
        var n = num
        var m = 0
        for (i in 1..32) {
            m = m shl 1
            m += n % 2
            n = n ushr 1
        }
        return m
    }
}